src/contrib/mlir/compiler/pass/mlir_subgraph_extraction.cpp

```diff
@@ -139,85 +133,50 @@ bool MLIRSubgraphExtractionPass::run_on_function(std::shared_ptr<Function> func)
 void MLIRSubgraphExtractionPass::build_subgraphs(std::shared_ptr<Function> func)
 {
     NGRAPH_DEBUG << "[CK Extract] Construct sub-graphs";
+    bool create_subgraph = true;
+    int current_subgraph_id = 0;
     for (auto op : func->get_ordered_ops())
```

Comment:
We need to have our own topological sort here to minimize sub-graphs. When multiple nodes are ready (all predecessors assigned a sub-graph), we need to pick the one that matches the last selected node. For example, if the last node assigned is an MLIR node, then we should pick another MLIR node to add to the sub-graph, if any is available. Similarily, if last node is unsupported, then we should pick and skip the unsupported node.

Reply:
Will change it.